array-reduce/main.js

```diff
@@ -0,0 +1,62 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sumReducer = (previousValue, currentValue) => previousValue + currentValue;
+const sum = numbers.reduce(sumReducer);
+console.log('The value of sum is: ', sum);
+
+const productReducer = (previousValue, currentValue) => previousValue * currentValue;
+const product = numbers.reduce(productReducer);
+console.log('The value of product is: ', product);
+
+const accounttReducer = (previousValue, currentValue) => {
+  let balance = 0;
+  if (isNaN(Number(previousValue))) {
+    if (previousValue.type === 'withdrawal') {
+      balance += (previousValue.amount * -1);
+    } else {
+      balance += previousValue.amount;
+    }
+  } else {
+    balance = previousValue;
+  }
+  if (currentValue.type === 'withdrawal') {
+    balance += (currentValue.amount * -1);
+  } else {
+    balance += currentValue.amount;
+  }
+  return balance;
+};
+const balance = account.reduce(accounttReducer);
```

Comment:
Good job getting this to work but a better approach is to start `previousValue` as `0` by passing `0` as a second argument to `reduce` that will avoid the extra checking if `previousNumber` is a number, it will always be a number.

```JAVASCRIPT
const balance = account.reduce(accounttReducer, 0);
```